src/backend/joanie/core/models/products.py

```diff
@@ -378,9 +378,32 @@ def can_be_state_validated(self):
         """
         return self.total == 0 or self.invoices.count() > 0
 
+    def can_be_state_submitted(self):
+        """
+        An order can be submitted if the order has a course, an organization,
+        an owner, and a product
```

Comment:
another condition is : if the order price is more than zero and the payment condition is credit card (always the case for the moment), then a credit card id must be filled on the order as well as an invoicing address.

Reply:
Currently we can handle a payment without card, do we want to delete this behaviour?